src/h5web/dataset-visualizer/VisDisplay.tsx

```diff
@@ -39,11 +40,18 @@ function VisDisplay(props: Props): ReactElement {
   useEffect(() => {
     scheduleLoadingFlag(); // in case retrieving value takes too long (e.g. initial fetch of `SilxProvider`)
 
-    getValue(dataset.id).then((val) => {
+    whileMounted(getValue(dataset.id)).then((val) => {
       cancelLoadingFlag();
       mergeState({ value: val, loading: false });
```

Comment:
In case this isn't clear, thanks to `whileMounted`, the `then` callback isn't called if the component has been unmounted. I had a warning in the tests.

Reply:
Thanks for the clarification. Mind putting this in comment in the code ? :sweat_smile:

Reply:
Doesn't feel right adding a comment for this technical detail. I was hoping that `whileMounted` would provide enough information. If anything I would rather switch to a more imperative solution like:

```ts
getValue(dataset.id)).then((val) => {
    if (isUnmounted) {
      return;
    }
    
    cancelLoadingFlag();
    mergeState({ value: val, loading: false });
    }
  });
```

Can we think of a better name for `whileMounted` maybe? `ifMounted`, `ifStillMounted`, `unlessUnmounted`, ...

Reply:
`ifMounted` seems good :+1: